### PR TITLE
MNT-22156 : [Security] Unauthenticated call to internal URLs using RSS feed in Share

### PR DESCRIPTION
**Pull request by alexbalmus**
Disable RSS dashlet by default + implement flag to enable it on demand.

The JVM flag will be: `-Drssdashlet.enabled=true`

### CODE REVIEW

File: share/src/main/java/org/alfresco/web/scripts/RssDashletStatusWebScript.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2005 - 2020 Alfresco Software Limited.
```
Comment (alexbalmus) [inline share/src/main/java/org/alfresco/web/scripts/RssDashletStatusWebScript.java:2]: The copyright year will need to be addressed as part of a separate task for all source files plus the license header text file used by the license check plugin.

**Review by skopf — CHANGES_REQUESTED** [review]
In this PR, I can only see the logic to hide the RSS dashlet in the UI. But the entire backend logic is still there.
The actual vulnerability however is in the backend logic which performs the requests on behalf of the user.
I assume there is a second part that goes with this commit that blocks the actual backend action? CC: @DavidCervigni

**Comment by alexbalmus** [comment]
@skopf, the backend logic is in async-rssfeed.get.js which has also been addressed in this PR.
Therefore, a request like the one presented in the Jira ticket description will no longer work (the "feed" JS object will only contain a field "error" with the content "RSS dashlet has been disabled."

**Review by skopf — APPROVED** [review]
Ah, of course. My bad. Should look closer at the path next time.

**Review by pieCit87 — APPROVED** [review]

**Pull request merged**